src/easylink/step.py

```diff
@@ -86,23 +89,33 @@ def update_implementation_graph(
         """Return a single node with an implementation attribute."""
         implementation_name = step_config[self.name]["implementation"]["name"]
         implementation_config = step_config[self.name]["implementation"]["configuration"]
+        implementation_node_name = (
```

Comment:
I'm using a simple heuristic for when we need to make the implementation node name more complicated than just the implementation name.

Reply:
Why not have it always be `f"{self.name}_{implementation_name}"`?

Reply:
It seemed to me that implementation names might often have the step in their names even in the "real" steps and implementations, and that it would be nice to avoid that redundancy except in cases where it is necessary to disambiguate duplicate implementations.

Reply:
> that implementation names might often have the step in their names even in the "real" steps

I think this is possible, especially when breaking up an existing software into multiple implementations (e.g. Splink into a blocking part, a matching part, etc.)